nominal/sdk.py

```diff
@@ -271,6 +279,108 @@ def _from_conjure(cls, client: NominalClient, dataset: scout_catalog.EnrichedDat
             _client=client,
         )
 
+@dataclass(frozen=True)
+class LogSet:
+    rid: str
+    created_by: str
+    name: str
+    created_at: datetime
+    updated_at: datetime
+    log_count: int
+    timestamp_type: DataSourceTimestampType
+    origin_metadata: Mapping[str, str]
+    description: str | None
+    _client: NominalClient = field(repr=False)
+
+    def attach_logs_and_finalize_request(self, logs: list[Log]) -> Self:
+        """
+        Attaches a list of logs to this log set and finalizes it. A logset is not considered readable
+        until it has been finalized. Once finalized, the logset is immutable.
+        """
+        request = datasource_logset_api.AttachLogsAndFinalizeRequest(
+            logs=[log._to_conjure() for log in logs],
+        )
+        response = self._client._logset_client.attach_logs_and_finalize(
+            auth_header = self._client._auth_header,
+            log_set_rid = self.rid,
+            request = request
+        )
+        log_set_metadata = self.__class__._from_conjure(self._client, response)
+        update_dataclass(self, log_set_metadata, fields=self.__dataclass_fields__)
+        return self
+    
+    def search_logs(
+            self,
+            page_size: int | None = None,
+            next_page_token: str | None = None,
+        ):
+        """
+        Search for logs within this log set.
+        """
+        request = datasource_logset_api.SearchLogsRequest(
+            token = next_page_token,
+            page_size = page_size,
+        )
+        response = self._client._logset_client.search_logs(
+            self._client._auth_header,
+            log_set_rid = self.rid,
+            request = request,
+        )
```

Comment:
For now the pagination functionality just mirrors conjure, AKA not very intuitive .. - I saw you were using yield for Runs and so maybe I should emulate that?